plugins/outputs/iotdb/iotdb.go

```diff
@@ -0,0 +1,287 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package iotdb
+
+// iotdb.go
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"math"
+
+	// Register IoTDB go client
+	"github.com/apache/iotdb-client-go/client"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+// DO NOT REMOVE THE NEXT TWO LINES! This is required to embed the sampleConfig data.
+//go:embed sample.conf
+var sampleConfig string
+
+type IoTDB struct {
+	Host            string `toml:"host"`
+	Port            string `toml:"port"`
+	User            string `toml:"user"`
+	Password        string `toml:"password"`
+	Timeout         int    `toml:"timeout"`
+	ConvertUint64To string `toml:"convertUint64To"`
+	TimeStampUnit   string `toml:"timeStampUnit"`
+	TreateTagsAs    string `toml:"treateTagsAs"`
+	session         *client.Session
+
+	Log telegraf.Logger `toml:"-"`
+}
+
+type RecordsWithTags struct {
+	// IoTDB Records basic data struct
+	DeviceIDList     []string
+	MeasurementsList [][]string
+	ValuesList       [][]interface{}
+	DataTypesList    [][]client.TSDataType
+	TimestampList    []int64
+	// extra tags
+	TagsList [][]*telegraf.Tag
+}
+
+func (*IoTDB) SampleConfig() string {
+	return sampleConfig
+}
+
+// Init is for setup, and validating config.
+func (s *IoTDB) Init() error {
+	var errorMsg string
+	if s.Timeout < 0 {
+		errorMsg = fmt.Sprintf("IoTDB Config Warning: The value of 'timeout' is negative:%d. Now it's fixed to 0.", s.Timeout)
+		s.Log.Warnf(errorMsg)
+		s.Timeout = 0
+	}
+	if !(s.ConvertUint64To == "ToInt64" ||
+		s.ConvertUint64To == "ForceToInt64" ||
+		s.ConvertUint64To == "Text") {
+		errorMsg = fmt.Sprintf("IoTDB Config Warning: The value of 'ConvertUint64To' is invalid: %s. Now it's fixed to 'ToInt64'.", s.ConvertUint64To)
+		s.Log.Warnf(errorMsg)
+		s.ConvertUint64To = "ToInt64"
+	}
```

Comment:
We have `internal/choice` for these kind of checks
```suggestion
	if !choice.Contains(s.ConvertUint64To, []string{"int64", "int64 clip", "string") {
		return fmt.Errorf("unknown 'uint64_conversion' method %q, s.ConvertUint64To)
	}
```
Please use the same scheme for the other checks below!

Reply:
This is cleaner and more beautiful! Thanks for suggestion. "choice.Contains" is a good friend.